proxy/gateway/java/gateway-remote-module-sdk/src/main/java/com/microsoft/azure/gateway/remote/ProxyGateway.java

```diff
@@ -0,0 +1,364 @@
+                        // Codes_SRS_JAVA_PROXY_GATEWAY_24_020: [ *Message Listener task - Create message* - If the Create message finished processing, it shall send an ok message to the Gateway. ]
+                        boolean sent = sendControlReplyMessage(RemoteModuleReplyCode.OK.getValue());
+                        // Codes_SRS_JAVA_PROXY_GATEWAY_24_021: [ *Message Listener task - Create message* - If ok message fails to be send to the Gateway, it shall do call module `destroy` and disconnect from message channel. ]
+                        if (!sent)
+                            this.disconnectDataMessage();
+                    } catch (ConnectionException e) {
+                        logger.error(e.toString());
+                        this.sendControlReplyMessage(RemoteModuleReplyCode.CONNECTION_ERROR.getValue());
+                    } catch (ModuleInstantiationException e) {
+                        logger.error(e.toString());
+                        this.sendControlReplyMessage(RemoteModuleReplyCode.CREATION_ERROR.getValue());
+                    }
+                }
+
+                if (controlMessage.getMessageType() == RemoteMessageType.START) {
+                    this.processStartMessage();
+                }
+
+                if (controlMessage.getMessageType() == RemoteMessageType.DESTROY) {
+                    this.processDestroyMessage();
+                }
+            }
+        }
+
+        void executeDataMessage() {
+            try {
+                // Codes_SRS_JAVA_PROXY_GATEWAY_24_025: [ *Message Listener task - Data message* - It shall not check for messages, if the message channel is not available. ]
+                if (this.dataEndpoint != null) {
+                    RemoteMessage dataMessage = this.dataEndpoint.receiveMessage();
+                    // Codes_SRS_JAVA_PROXY_GATEWAY_24_027: [ *Message Listener task - Data message* - If no data message is received or if an error occurs, it shall do nothing. ]
+                    if (dataMessage != null) {
+                        // Codes_SRS_JAVA_PROXY_GATEWAY_24_026: [ *Message Listener task - Data message* - If data message is received, it shall forward it to the module by calling `receive` method. ]
+                        this.module.receive(((DataMessage) dataMessage).getContent());
+                    }
+                }
+            } catch (ConnectionException e) {
+                logger.error(e.toString());
+            } catch (MessageDeserializationException e) {
+                logger.error(e.toString());
+            }
+        }
+
+        private void processDestroyMessage() {
+            this.detach(false);
+        }
+
+        private void processStartMessage() {
+            if (this.module == null)
+                return;
+
+            // Codes_SRS_JAVA_PROXY_GATEWAY_24_022: [ *Message Listener task - Start message* - If message type is START, it shall call module `start` method. ]
+            this.module.start();
```

Comment:
no need to check because the module has to implement IGatewayModule which has start.